bpf/tests/pid_match_test.go

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package bpf
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/tetragon/pkg/api/processapi"
+	"github.com/cilium/tetragon/pkg/k8s/apis/cilium.io/v1alpha1"
+	"github.com/cilium/tetragon/pkg/selectors"
+	"github.com/cilium/tetragon/pkg/sensors/exec/execvemap"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const programNamePidMatch = "test_pid_match"
+
+type testContext struct {
+	coll      *ebpf.Collection
+	prog      *ebpf.Program
+	execvMap  *ebpf.Map
+	resultMap *ebpf.Map
+	t         *testing.T
+}
+
+func setupTest(t *testing.T) (*testContext, error) {
+	ctx := &testContext{t: t}
+
+	coll, err := ebpf.LoadCollection("objs/pid_match_test.o")
+	if err != nil {
+		var ve *ebpf.VerifierError
+		if errors.As(err, &ve) {
+			return nil, fmt.Errorf("verifier error: %+v", ve)
+		}
+		return nil, err
+	}
+	ctx.coll = coll
+
+	var ok bool
+	ctx.prog, ok = coll.Programs[programNamePidMatch]
+	require.True(t, ok, "program %s not found", programNamePidMatch)
+
+	ctx.execvMap, ok = coll.Maps["execve_map"]
+	require.True(t, ok, "execve_map not found")
+
+	ctx.resultMap, ok = coll.Maps["test_result_map"]
+	require.True(t, ok, "test_result_map not found")
+
+	return ctx, nil
+}
+
+func (ctx *testContext) cleanup() {
+	if ctx.coll != nil {
+		ctx.coll.Close()
+	}
+}
+
+func (ctx *testContext) runProg(selfPid uint32) error {
+	err := ctx.execvMap.Update(uint32(0), &execvemap.ExecveValue{
+		Process: processapi.MsgExecveKey{Pid: selfPid},
+	}, 0)
+	if err != nil {
+		return fmt.Errorf("failed to update execve map: %w", err)
+	}
+	_, err = ctx.prog.Run(&ebpf.RunOptions{})
+	return err
```

Comment:
why not have test_pid_match to return the err and take get it in here,
and you could drop test_result_map map